src/main/java/winstone/Logger.java

```diff
@@ -6,160 +6,61 @@
-            OutputStream defaultStream,
-            Charset defaultStreamCharset,
             boolean showThrowingThreadArg) {
         synchronized (semaphore) {
             if (!initialised) { // recheck in case we were blocking on another init
-                initialised = false;
-                LOGGER.setLevel(level);
-                namedStreams = new HashMap<>();
-//                nullStreams = new ArrayList();
-                initialised = true;
-                setStream(DEFAULT_STREAM, defaultStream, defaultStreamCharset);
+                java.util.logging.Logger.getLogger("").setLevel(level);
                 showThrowingThread = showThrowingThreadArg;
+                initialised = true;
             }
         }
     }
 
-    /**
-     * Allocates a stream for redirection to a file etc
-     *
-     * @deprecated Use {@link #setStream(String, OutputStream, Charset)}
-     */
-    @Deprecated
-    public static void setStream(String name, OutputStream stream) {
-        setStream(name, stream, Charset.defaultCharset());
-    }
-
-    /**
-     * Allocate a stream for redirection to a file etc
-     */
-    public static void setStream(String name, OutputStream stream, Charset charset) {
-        setStream(name, stream != null ? new OutputStreamWriter(stream, charset) : null);
-    }
-
-    /**
-     * Allocates a stream for redirection to a file etc
-     */
-    @SuppressFBWarnings(value = "EI_EXPOSE_STATIC_REP2", justification = "False positive, this is intended")
-    public static void setStream(String name, Writer stream) {
-        if (name == null) {
-            name = DEFAULT_STREAM;
-        }
-        if (!initialised) {
-            init(INFO);
-        }
-        synchronized (semaphore) {
-            if (name.equals(DEFAULT_STREAM)) {
-                defaultStream = stream;
-            } else if (stream == null) {
-                namedStreams.remove(name);
-            } else {
-                namedStreams.put(name, stream);
-            }
-        }
-    }
-
-    /**
-     * Forces a flush of the contents to file, display, etc
-     */
-    public static void flush(String name) {
-        if (!initialised) {
-            init(INFO);
-        }
-
-        Writer stream = getStreamByName(name);
-        if (stream != null) {
-            try {stream.flush();} catch (IOException err) {}
-        }
-    }
-
-    private static Writer getStreamByName(String streamName) {
-        if ((streamName != null) && streamName.equals(DEFAULT_STREAM)) {
-            // As long as the stream has not been nulled, assign the default if not found
-            synchronized (semaphore) {
-                Writer stream = (Writer) namedStreams.get(streamName);
-                if ((stream == null) && !namedStreams.containsKey(streamName)) {
-                    stream = defaultStream;
-                }
-                return stream;
-            }
-        } else {
-            return defaultStream;
-        }
-
-    }
-
```

Comment:
All of this streams-related logic was dead code that was setting fields but never getting them. Deleting it makes it more clear what is going on in this class, which is really not very much other than delegating to JUL.